config.go

```diff
@@ -1372,13 +1373,75 @@ func extractBitcoindRPCParams(bitcoindConfigPath string) (string, string, string
 }
 
 // checkZMQOptions ensures that the provided addresses to use as the hosts for
-// ZMQ rawblock and rawtx notifications are different.
+// ZMQ rawblock and rawtx notifications are valid and different.
 func checkZMQOptions(zmqBlockHost, zmqTxHost string) error {
-	if zmqBlockHost == zmqTxHost {
-		return errors.New("zmqpubrawblock and zmqpubrawtx must be set" +
+	// ZMQ can be used with different transports
+	// (http://api.zeromq.org/4-0:zmq-bind). At first, parse the socket URL
+	// and perform generalized comparison.
+	zmqBlockURL, err := url.Parse(zmqBlockHost)
+	if err != nil {
+		return err
+	}
+
+	zmqTxURL, err := url.Parse(zmqTxHost)
+	if err != nil {
+		return err
+	}
+
+	if zmqBlockURL == zmqTxURL {
+		return errors.New("zmqpubrawblock and zmqpubrawtx must be set " +
 			"to different addresses")
 	}
 
+	// TCP transport specific checks.
+	if zmqBlockURL.Scheme == "tcp" && zmqTxURL.Scheme == "tcp" {
+		// Ensure the host:port is correct and resolve it to ip:port if needed,
+		// then split the result to IP and port
+		zmqBlockResolved, err := net.ResolveTCPAddr("tcp", zmqBlockURL.Host)
+		if err != nil {
+			return err
+		}
+		zmqBlockHost, zmqBlockPort, err := net.SplitHostPort(
```

Comment:
Don't think there's a need for this to retrieve the IP and port. `ResolveTCPAddr` returns a `TCPAddr`, which already exposes the IP and port as fields.

Reply:
Oh yeah, you're right! Thanks, fixed.